releasenotes/notes/64-bit-only-4132f330ec7804b3.yaml

```diff
@@ -0,0 +1,14 @@
+---
+upgrade:
+  - |
+    Qiskit no longer supports Linux i686 and 32 bit Windows. Starting in Qiskit
+    2.0 a 64 bit platform is needed to run Qiskit. The user base for 32bit
+    architectures is relatively small and as Qiskit continues to focus on
+    improving performance to handle the increased scale in the complexity
+    of quantum computing hardware maintaining support for 32 bit platforms
+    is proving increasingly difficult. This coupled with the larger
+    scientific/numeric Python community's trend away from 32 bit platform
+    support maintaining support for 32bit platforms is no longer viable.
+    Qiskit 1.x will still continue to support 32 bit platforms, but starting
+    in this 2.0.0 release Qiskit no longer supports these platforms, and will
+    not publish pre-compiled binaries for these platforms any longer.
```

Comment:
On the last sentence: do we want to drive home even further that we will not even commit to Qiskit _building_ on 32-bit platforms at all?  Your note does already imply it, in some ways.

In particular, I fairly strongly expect that as soon as we commit to dropping 32-bit support, I'll rejig `PackedOperation` a little bit to explicitly store a pointer-like thing within itself rather than a `u64` (assuming #13486), so we can start moving in the Miri-checkable direction of strict pointer providence, and at that point we'll be requiring that the width of `*mut ()` is 64 bits.

Reply:
When I wrote this I thought about that, but I didn't want to call it out explicitly because we can still build on 32bit platforms right now and I wasn't 100% sure we would land the changes in 2.0 that would break this. But I can say something like "and we're no longer providing any guarantee of being able to build from source on 32 bit platforms."

Reply:
```suggestion
    Qiskit 1.x will still continue to support 32 bit platforms, but starting
    in this 2.0.0 release Qiskit no longer supports these platforms, will
    not publish pre-compiled binaries for these platforms any longer, and
    there is no longer any guarantee of being able to build from source on
    32 bit platforms.
```